runtime/flang/mmul_real16str1.F95

```diff
@@ -0,0 +1,811 @@
+!
+! Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+! See https://llvm.org/LICENSE.txt for license information.
+! SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+!
+
+#include "mmul_dir.h"
+
+subroutine F90_matmul_real16_str1(dest,s1,s2, &
+      k_extnt,m_extnt,n_extnt,                   &
+      s1_d1_extnt,s2_d1_extnt,d_d1_extnt,        &
+      d_d1_lstride)
+
+      DESC_INT  n_extnt,m_extnt,k_extnt
+      DESC_INT  s1_d1_extnt,s2_d1_extnt,d_d1_extnt,d_d1_lstride
+      INTEGER   bs
+#if defined(LINUX8664)
```

Comment:
what about other OS'es and CPU archs?

Reply:
This code, including the use of `LINUX8664` macro, was modeled after (copied from) `mmul_real8str1.F95`. If the macro is set, it will cause the code to call `_F90_matmul_real16_str1a`/`_F90_matmul_real8_str1a`. These functions are actually not implemented, and the `LINUX8664` macro is only defined in `runtime/libpgmath/lib/CMakeLists.txt`, and is not defined when compiling `runtime/flang/*`. I think it is probably okay to delete this macro and the code it guards.